crates/compiler/mono/src/drop_specialization.rs

```diff
@@ -1490,3 +1498,60 @@ fn low_level_no_rc(lowlevel: &LowLevel) -> RC {
         }
     }
 }
+
+#[derive(Clone)]
+struct CountingMap<K>
+where
+    K: Eq + std::hash::Hash + Clone,
+{
+    map: MutMap<K, u64>,
+}
+
+impl<K> CountingMap<K>
+where
+    K: Eq + std::hash::Hash + Clone,
+{
+    fn new() -> Self {
+        Self {
+            map: MutMap::default(),
+        }
+    }
+
+    fn insert(&mut self, key: K) {
+        self.insert_count(key, 1);
+    }
+
+    fn insert_count(&mut self, key: K, count: u64) {
+        self.map
+            .entry(key)
+            .and_modify(|c| *c += count)
+            .or_insert(count);
+    }
+
+    fn pop(&mut self, key: &K) -> bool {
+        match self.map.get_mut(key) {
+            Some(0) => false,
+            Some(c) => {
+                *c -= 1;
+                true
+            }
+            None => false,
+        }
+    }
+
+    fn contained(&self, symbol: &K) -> bool {
+        self.map.contains_key(symbol)
+    }
```

Comment:
this reads like a past tense. why is this better than `contains` (which is much more standard in the ecosystem)?